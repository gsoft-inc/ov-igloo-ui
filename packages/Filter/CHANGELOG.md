# @igloo-ui/filter

## 0.3.2

### Patch Changes

- 6ea531f: Updated dependency versions for react-aria packages, luxon and hopper

## 0.3.1

### Patch Changes

- 5c4c95e: Fix of filter border color when disabled

## 0.3.0

### Minor Changes

- 4846c59: Wrapped Igloo CSS in @layer to isolate it.

## 0.2.3

### Patch Changes

- cfb5619: Updated hopper token versions and button versions.

## 0.2.2

### Patch Changes

- 7d36815: Updated hopper token versions and igloo icon versions.

## 0.2.1

### Patch Changes

- a2df55d: Updated to latest Hopper Token dependency

## 0.2.0

### Minor Changes

- 9739155: Updated storybook and recharts. Applied the new brand look for filter.

### Patch Changes

- 9739155: Gave ColorPicker the new Wokleap look. Color now uses css variables to update its color. Fixed spacing issues in tooltip. Updated hopper token version.

## 0.1.4

### Patch Changes

- bf4ddef: Updated storybook version and linting. Made linting fixes in these components.

## 0.1.3

### Patch Changes

- a41e1d4: Fix: moved destructuring of props to function parameters so that storybook will automatically show the default value in the props table. It was done in Modal because it needed to send props to useDialog. In this case, @default annotation was used in comments.

## 0.1.2

### Patch Changes

- 6ee9c45: Updated anything causing errors or warnings in the browser console

## 0.1.1

### Patch Changes

- 0640409: udpate classnames to 2.3.2

## 0.1.0

### Minor Changes

- 2f042c6: Feat: initial release of the filter component
