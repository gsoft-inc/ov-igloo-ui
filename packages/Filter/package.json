{
  "name": "@igloo-ui/filter",
  "version": "0.3.1",
  "main": "dist/Filter.js",
  "module": "dist/Filter.js",
  "types": "dist/Filter.d.ts",
  "styles": "dist/filter.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Vicky Comeau vicky.comeau@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c rollup.config.js"
  },
  "dependencies": {
    "@igloo-ui/tokens": "^2.0.0",
    "@hopper-ui/tokens": "^3.1.3",
    "classnames": "^2.3.2"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
