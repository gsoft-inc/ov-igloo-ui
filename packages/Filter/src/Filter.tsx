import * as React from 'react';

import cx from 'classnames';
import './filter.scss';

export interface FilterProps extends React.ComponentProps<'button'> {
  /** Add the main text/icon to the filter */
  children: React.ReactNode;
  /** Add class names to the filter button */
  className?: string;
  /** Count of items that match the filter */
  count?: number;
  /** Add a data-test tag for automated tests */
  dataTest?: string;
  /** True if the filter should be disabled */
  disabled?: boolean;
  /** Add an event for when the filter is clicked */
  onClick?: () => void;
  /** True if the tag is selected */
  selected?: boolean;
}

const Filter: React.FunctionComponent<FilterProps> = (props: FilterProps) => {
  const {
    children,
    className,
    count,
    dataTest,
    disabled,
    onClick,
    selected,
    ...rest
  } = props;

  const classes = cx('ids-filter', className, {
    'ids-filter--selected': selected,
  });

  return (
    <button
      type="button"
      className={classes}
      onClick={onClick}
      data-test={dataTest}
      disabled={disabled}
      {...rest}
    >
      {children}
      {count && <span className="ids-filter__count">{count}</span>}
    </button>
  );
};

export default Filter;
