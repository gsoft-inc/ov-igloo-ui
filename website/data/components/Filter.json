[
  [
    {
      "tags": {},
      "filePath": "/Users/franck.gaudin/Devel/Igloo/components/packages/Filter/src/Filter.tsx",
      "description": "",
      "displayName": "Filter",
      "methods": [],
      "props": {
        "children": {
          "defaultValue": null,
          "description": "Add the main text/icon to the filter",
          "name": "children",
          "parent": {
            "fileName": "components/packages/Filter/src/Filter.tsx",
            "name": "FilterProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Filter/src/Filter.tsx",
              "name": "FilterProps"
            },
            {
              "fileName": "components/node_modules/@types/react/index.d.ts",
              "name": "TypeLiteral"
            }
          ],
          "required": false,
          "type": { "name": "ReactNode" }
        },
        "className": {
          "defaultValue": null,
          "description": "Add class names to the filter",
          "name": "className",
          "parent": {
            "fileName": "components/packages/Filter/src/Filter.tsx",
            "name": "FilterProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Filter/src/Filter.tsx",
              "name": "FilterProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "dataTest": {
          "defaultValue": null,
          "description": "Add a data-test tag for automated tests",
          "name": "dataTest",
          "parent": {
            "fileName": "components/packages/Filter/src/Filter.tsx",
            "name": "FilterProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Filter/src/Filter.tsx",
              "name": "FilterProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "disabled": {
          "defaultValue": null,
          "description": "True if the filter should be disabled",
          "name": "disabled",
          "parent": {
            "fileName": "components/packages/Filter/src/Filter.tsx",
            "name": "FilterProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Filter/src/Filter.tsx",
              "name": "FilterProps"
            }
          ],
          "required": false,
          "type": { "name": "boolean" }
        },
        "onClick": {
          "defaultValue": null,
          "description": "Add an event for when the filter is clicked",
          "name": "onClick",
          "parent": {
            "fileName": "components/packages/Filter/src/Filter.tsx",
            "name": "FilterProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Filter/src/Filter.tsx",
              "name": "FilterProps"
            }
          ],
          "required": false,
          "type": { "name": "() => void" }
        },
        "selected": {
          "defaultValue": null,
          "description": "True if the tag is selected",
          "name": "selected",
          "parent": {
            "fileName": "components/packages/Filter/src/Filter.tsx",
            "name": "FilterProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Filter/src/Filter.tsx",
              "name": "FilterProps"
            }
          ],
          "required": false,
          "type": { "name": "boolean" }
        }
      }
    }
  ]
]
