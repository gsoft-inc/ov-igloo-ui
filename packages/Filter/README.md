# Filter

The filter component is a button that is used to filter the items of a list or table.

<Example is="custom" />
<ReferenceLinks is="custom" />

## Installation

To install `@igloo-ui/filter` in your project, you will need to run the following command using [npm](https://www.npmjs.com/):

```bash
npm install @igloo-ui/filter
```

If you prefer [Yarn](https://classic.yarnpkg.com/en/), use the following command instead:

```bash
yarn add @igloo-ui/filter
```

## Usage

Then to use the component in your code just import it!

```jsx
import Filter from '@igloo-ui/filter';

<Filter
  onClick={() => {
    setSelected(!selected);
    alert('Filter was clicked');
  }}
  selected={selected}
>
  Assigned to me (5)
</Filter>;
```
