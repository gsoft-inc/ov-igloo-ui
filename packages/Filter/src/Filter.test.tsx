/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';

import Filter from './Filter';

describe('Filter', () => {
  test('It should render a snapshot', () => {
    const { asFragment } = render(
      <Filter dataTest="filter1">Assigned to me</Filter>
    );
    expect(screen.getByTestId('filter1')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('It should render a disabled state', () => {
    render(
      <Filter dataTest="filter2" disabled>
        Assigned to me
      </Filter>
    );
    const filter = screen.getByTestId('filter2');
    expect(filter).toBeDisabled();
  });

  test('It should render a selected look', () => {
    render(
      <Filter dataTest="filter3" selected>
        Assigned to me
      </Filter>
    );
    const filter = screen.getByTestId('filter3');
    expect(filter).toHaveClass('ids-filter--selected');
  });

  test('It should render a filter with a count', () => {
    render(
      <Filter dataTest="filter4" count={4}>
        Assigned to me
      </Filter>
    );
    const filter = screen.getByTestId('filter4');
    expect(filter).toContainHTML('<span class="ids-filter__count">4</span>');
  });
});
