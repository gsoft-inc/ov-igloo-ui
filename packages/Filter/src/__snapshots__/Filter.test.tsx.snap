// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter It should render a snapshot 1`] = `
<DocumentFragment>
  <button
    class="ids-filter"
    data-test="filter1"
    type="button"
  >
    Assigned to me
  </button>
</DocumentFragment>
`;
